### ISSUE TITLE
Useful information in the banners

### ISSUE THREAD

**Issue by return42** [searxng/searxng#3176]
We have various requests/comments that can be roughly summarized under the term "banner".

We should think about how we could implement such a banner.  Please use this thread and let us know your suggestions.

What have already been said about:

- https://github.com/searxng/searxng/pull/3135#issuecomment-1915611622
- https://github.com/searxng/searxng/pull/3135#issuecomment-1920996935
- https://github.com/searxng/searxng/issues/3156
- https://github.com/iv-org/invidious/issues/2166 (from the invidious project)

----

my, still unfinished thoughts:

- I could imagine that there is a banner in which useful information for the user can be displayed. 
- Whether this banner is placed in the upper or better in the lower area would still have to be evaluated (UI).
- A text can be configured for this banner, this text is set in Markdown format in the configuration.
- This banner can also be used by SearXNG and the plugins to provide the user with additional (context-related) information.
- If there is no context-related information, the configured text is displayed by default
- If no text is configured, the name of the instance is displayed (#3156)

**Comment by perennialtech** [comment]
Agreed that this topic warrants a separate issue for further discussion. Given that we don't have any visual examples/concepts to go off of, my comments are similarly speculative.

> Whether this banner is placed in the upper or better in the lower area would still have to be evaluated (UI).

If the banner will be placed near the top of the page, viz. the implementation by Invidious, it should be visible by default, but hideable by the user. Once it's read, a section containing information not directly related to what the user is doing is annoying and unwieldy to navigate around, especially if the banner is large. From my understanding, this feature would require additional JavaScript and a cookie for the preference to persist across different pages.

On the other hand, if the banner is placed at the bottom of the page, such an implementation would be less intrusive and avoid including additional JS. However, it would be less likely to be read except on a select few pages (e.g., home, about, and preferences).

> A text can be configured for this banner, this text is set in Markdown format in the configuration.

Using Markdown for the banner is a good compromise between versatility and security compared with HTML, and I would be fine with such an implementation.

> This banner can also be used by SearXNG and the plugins to provide the user with additional (context-related) information.

> If there is no context-related information, the configured text is displayed by default

Perhaps this could be split into a separate (but closely related) UI element, since we can't really control for how much content instance maintainers will add to the banner and this could lead to inconsistency.

> If no text is configured, the name of the instance is displayed

With the current styling of simple theme emphasising padding between UI elements (compared with the Oscar theme where elements are more compact), this might be difficult to implement gracefully without creating excessive empty space on the page.

**Comment by return42** [comment]
>  especially if the banner is large.

I had a one-liner in mind .. directly above the search field:

![image](https://github.com/searxng/searxng/assets/554536/3df83240-63d6-45fa-848d-8d4aa6ef2690)

that one-liner can link to further information / if needed.

Whether the one-liner is a treadmill or an opening box (mouse-over) would have to be decided.

We will need JS for this ... if JS is disabled the one-liner will not be displayed(?). Or does the one-liner always have to be displayed?

> this might be difficult to implement gracefully without creating excessive empty space on the page.

What I want to avoid are multi-lines which tend to be intrusive / especially on a small device :

![image](https://github.com/searxng/searxng/assets/554536/04875d12-36e3-4488-826c-e9a3bc944d20)

>> This banner can also be used by SearXNG and the plugins to provide the user with additional (context-related) information.
>> If there is no context-related information, the configured text is displayed by default

> Perhaps this could be split into a separate (but closely related) UI element, 

True, we can decide it later when we know how the UI element (of the one-liner) will look like ..

**Comment by perennialtech** [comment]
> I had a one-liner in mind .. directly above the search field:

This might look out of place for the simple theme, since search engines with modern styling usually have the logo and search bar as the sole top-most elements.

![duckduckgo](https://github.com/searxng/searxng/assets/139829103/8339fbb6-63b9-4249-a60d-6a15c9c12fed)

On the other hand, there aren't many alternative locations for such a banner and placement directly above the search bar would likely be the best choice.

A fixed footer might be considered as an alternative, but if JS is disabled, the banner would consistently take up a large amount of space as the entire banner would need to be displayed.

![nytimes](https://github.com/searxng/searxng/assets/139829103/a2bea084-b86f-48c0-b1a9-0a81f7b9ff08)

> Whether the one-liner is a treadmill or an opening box (mouse-over) would have to be decided.

An [accordion element](https://getbootstrap.com/docs/5.3/components/accordion/) might be the better choice as they are more commonly used to condense content and allow users to expand it to see more information.

> We will need JS for this ... if JS is disabled the one-liner will not be displayed(?). Or does the one-liner always have to be displayed?

> What I want to avoid are multi-lines which tend to be intrusive / especially on a small device :

In the case of JS being disabled, the banner should be displayed in its entirety at all times. This would be a UI compromise, but a better alternative versus the information being entirely absent. 

A mobile device that has JS disabled or unavailable is rare nowadays, so I don't consider a multi-line banner taking up excessive space on a small device to be a concerning issue and would be addressed by the use of an accordion element.

**Comment by winrg** [comment]
`<details>` can be used to create js-free accordion elements.
